src/Adapter/Image/ImageRetriever.php

```diff
@@ -190,6 +199,48 @@ public function getImage($object, $id_image)
                 );
             }
 
+            $generateAdditionalWebP = (bool) Configuration::get('PS_ADDITIONAL_IMAGE_WEBP');
+
+            if ($generateAdditionalWebP) {
+                $resizedImagePathWebP = implode(DIRECTORY_SEPARATOR, [
+                    $imageFolderPath,
+                    sprintf('%s-%s.webp', $id_image, $image_type),
+                ]);
+
+                if (!file_exists($resizedImagePathWebP)) {
+                    ImageManager::resize(
+                        $originalImagePath,
+                        $thumbnailPath,
+                        (int) $image_type['width'],
+                        (int) $image_type['height'],
+                        'webp',
+                        true
+                    );
+                }
+
+                $additionalSources['webp'] = $this->link->$getImageURL($rewriteLink, $id_image, $image_type['name'], '.webp');
+            }
+
+            if ($this->avifExtensionChecker !== null && $this->avifExtensionChecker->isAvailable()) {
```

Comment:
we should not call avifExtensionChecker at image generation, it should be only for the display of the checkbox (the check is a little expensive and if the merchant checked avif, then it means it's available.